packages/blockchain/bnb/src/lib/services/bnb.tx.ts

```diff
@@ -0,0 +1,136 @@
+import { BnbApiCallsType } from '../../index'
+import { ApiServices, Currency, OpenAPI, TransferBnb } from '@tatumio/api-client'
+import { BncClient } from '@binance-chain/javascript-sdk'
+import { BigNumber } from 'bignumber.js'
+import { SDKArguments, SdkErrorCode } from '@tatumio/shared-abstract-sdk'
+import { decodeAddress, getAddressFromPrivateKey } from '@binance-chain/javascript-sdk/lib/crypto'
+import { AminoPrefix } from '@binance-chain/javascript-sdk/lib/types'
+import { BnbSdkError } from '../bnb.sdk.errors'
+
+export const bnbTxService = (args: SDKArguments, apiCalls: BnbApiCallsType) => {
+  const getMsgData = (
+    amount: string,
+    key: string,
+    address: string,
+    currency: string,
+    testnet: boolean,
+    from?: string,
+  ) => {
+    const fromAddress = from || getAddressFromPrivateKey(key, testnet ? 'tbnb' : 'bnb')
+    const accCode = decodeAddress(fromAddress)
+    const toAccCode = decodeAddress(address)
+    const coin = {
+      denom: currency,
+      amount: Math.round(new BigNumber(amount).multipliedBy(new BigNumber(10).pow(8)).toNumber()),
+    }
+
+    return {
+      fromAddress,
+      msg: {
+        inputs: [
+          {
+            address: accCode,
+            coins: [coin],
+          },
+        ],
+        outputs: [
+          {
+            address: toAccCode,
+            coins: [coin],
+          },
+        ],
+        aminoPrefix: AminoPrefix.MsgSend,
+      },
+      signMsg: {
+        inputs: [
+          {
+            address: fromAddress,
+            coins: [coin],
+          },
+        ],
+        outputs: [
+          {
+            address,
+            coins: [coin],
+          },
+        ],
+      },
+    }
+  }
+
+  const sendTransaction = async (
+    body: TransferBnb,
+    options: { testnet: boolean; token?: string; issuerAccount?: string },
+  ) => {
+    return ApiServices.blockchain.bnb.bnbBroadcast({
```

Comment:
you should wrap with if (body.signatureId) ApiServices.blockchain.bnb.sendTransaction) else this